air/src/utils.rs

```diff
@@ -1,8 +1,172 @@
 use alloc::vec::Vec;
-use core::ops::Range;
+use core::ops::{Add, AddAssign, Index, IndexMut, Mul, Range, Rem, Shr, Sub};
 
 use super::FieldElement;
-use vm_core::utils::range as create_range;
+use vm_core::{utils::range as create_range, Felt};
+
+// NEWTYPE WRAPPERS
+// ================================================================================================
+
+/// A newtype wrapper around a usize value representing a step in the execution trace.
+#[derive(Debug, Copy, Clone)]
+pub struct Step(u32);
+
+// FROM STEP
+
+impl From<Step> for u32 {
+    fn from(step: Step) -> u32 {
+        step.0
+    }
+}
+
+impl From<Step> for i32 {
+    fn from(step: Step) -> i32 {
+        step.0 as i32
+    }
+}
```

Comment:
This is not guaranteed to work, since the size of `u32` is larger than `i32`. Is there a reason we need to support conversions to `i32` at all?